source/jormungandr/jormungandr/scenarios/experimental.py

```diff
@@ -595,172 +51,97 @@ def call_kraken(self, request_type, request, instance, krakens_call):
         logger = logging.getLogger(__name__)
         logger.debug('datetime: %s', request['datetime'])
 
-        # odt_stop_points is a set of stop_point.uri with is_zonal = true used to manage tad_zonal
-        odt_stop_points = set()
-
-        # crowfly_stop_points is a set of stop_point.uri used to create a crowfly section.
-        crowfly_stop_points = set()
+        requested_dep_modes = {mode for mode, _ in krakens_call}
+        requested_arr_modes = {mode for _, mode in krakens_call}
+        res = []
 
-        if not g.requested_origin:
-            g.requested_origin = instance.georef.place(request['origin'])
-            if not g.requested_origin:
-                r = self._make_error_response("The entry point: {} is not valid".format(request['origin']),
-                                              response_pb2.Error.unknown_object)
-                return [r]
+        logger.debug('requesting from by uri: %s', request['origin'])
+        logger.debug('requesting to by uri: %s', request['destination'])
 
-        if not g.requested_destination:
-            g.requested_destination = instance.georef.place(request['destination'])
-            if not g.requested_destination:
-                r = self._make_error_response("The entry point: {} is not valid".format(request['destination']),
-                                              response_pb2.Error.unknown_object)
-                return [r]
+        requested_orig = PlaceByUri(instance, request['origin'])
+        requested_dest = PlaceByUri(instance, request['destination'])
 
-        worker = AsyncWorker(instance, krakens_call, request)
+        requested_orig_obj = requested_orig.wait_and_get()
```

Comment:
I do want to wait here to make sure that `from` and `to` are valid before launching complicated computation